modules/vault/main.tf

```diff
@@ -18,6 +18,7 @@ data "aws_caller_identity" "current" {}
 
 data "aws_iam_user" "vault_user" {
   user_name = var.vault_user == "" ? aws_iam_user.jenkins-x-vault[0].name : var.vault_user
+  depends_on = [aws_iam_user.jenkins-x-vault]
```

Comment:
nice digging with https://github.com/hashicorp/terraform/issues/15285#issuecomment-447971852 :-)
One would assume that the explicit depends_on is not needed, but the above comment explains things quite well.